planner_cspace/src/planner_3d.cpp

```diff
@@ -488,18 +513,23 @@ class Planner3dNode
             }
             cost += euclidCost(d, ec_rough_);
 
-            const float gp = c + cost;
-            if (gnext > gp)
-            {
-              gnext = gp;
+            const float cost_next = c + cost;
+            if (gnext > cost_next)
+              updates.push_back(GridmapUpdate(next, cost_next));
+          }
+        }
 #pragma omp critical
-              {
-                open.push(Astar::PriorityVec(gp, gp, next));
-              }
+        {
+          for (const GridmapUpdate& u : updates)
```

Comment:
update gridmap after leaving parallel section. this should avoid generating loop in search result tree.